spark/src/main/kotlin/com/adevinta/spark/res/AnnotatedStringResource.kt

```diff
@@ -291,16 +297,25 @@ private fun CharSequence.asAnnotatedString(
 ): AnnotatedString {
     if (this !is Spanned) return AnnotatedString(this.toString())
     return buildAnnotatedString {
-        append(this@asAnnotatedString.toString())
+        var annotatedString = this@asAnnotatedString
         getSpans(0, length, Any::class.java).forEach {
             val start = getSpanStart(it)
             val end = getSpanEnd(it)
-            buildWithSpan(it, start, end, density, colors, typography)
+            annotatedString = annotatedString.buildWithAccessibilitySpan(it, start, end)
+        }
+        append(annotatedString)
+        getSpans(0, length, Any::class.java).forEach {
+            val start = getSpanStart(it)
+            val end = getSpanEnd(it)
+            buildWithSpanStyle(it, start, end, density, colors, typography)
```

Comment:
I'm not sure to fully understand why we duplicated this part of the code. Can't we merge the tts handling with the already existing `buildWithSpan` which is already parsing the entire string/annotations once?

Reply:
That's the frustrating part, TtsSpan in spannable uses metadata to add info for Tts to better read the words but this is not possible yet in compose and AnnotatedStrings (well only Verbatim is possible for now)

So the only way I found was to actually reformat the string itself to make it more readable by Tts.

The first part with `buildWithAccessibilitySpan` updates the strings before using `append`, and the second part with `buildWithSpanStyle` only adds styling to the annotated string (and so needs to be done after the `append`)

But if you know how to extend the TtsAnnotation from Compose, or a better way to do this I'm all ears 🙏